### PR TITLE
Update grammar etc and examples in Components & Tags guide

### PR DESCRIPTION
**Pull request by dragoncoder047**
also, couldn't resist adding a little Dr. Who reference on line 180

### CODE REVIEW

File: guides/en/1_getting_started/3_components.md [hunk 1]
```diff
@@ -22,11 +22,11 @@ The components system of KAPLAY (the `C` of ECS) is powerful and flexible.
 ## What is a Component?
 
 A component is a piece of code that defines a specific behavior of a game
-object. It usually return a set of **properties** and **methods** that are
+object. It usually defines a set of **properties** and **methods** that are
```
Comment (lajbel) [inline guides/en/1_getting_started/3_components.md:25]: I prefer return word, is more like what it really does, returns something.

File: guides/en/1_getting_started/3_components.md [hunk 2]
```diff
@@ -22,11 +22,11 @@ The components system of KAPLAY (the `C` of ECS) is powerful and flexible.
...
 attached to the game object.
 
-For example, the [`pos()`](/doc/ctx/pos) component returns the `pos` property
-and the `move()` method:
+For example, the [`pos()`](/doc/ctx/pos) component gives the object a `pos` property
+and a `move()` method:
```
Comment (lajbel) [inline guides/en/1_getting_started/3_components.md:29]: It only returns one `move` and `pos`, so the

File: guides/en/1_getting_started/3_components.md [hunk 3]
```diff
@@ -39,19 +39,23 @@ console.log(player.pos); // { x: 180, y: 80 }, this is a property
 
 There are a lot of useful components in KAPLAY like:
 
-- [`sprite()`](/doc/ctx/sprite) for rendering images
-- [`area()`](/doc/ctx/area) for collision detection, with methods like
+- [`sprite()`](/doc/ctx/sprite) to give the object an image appearance, and play animations
```
Comment (lajbel) [inline guides/en/1_getting_started/3_components.md:42]: image appeareance sounds strange for me, maybe only image?

File: guides/en/1_getting_started/3_components.md [hunk 4]
```diff
@@ -39,19 +39,23 @@ console.log(player.pos); // { x: 180, y: 80 }, this is a property
...
 ## What is a tag?
 
-Tags are names, labels or keywords that group game objects, such as enemies,
-friends, etc.
+A tag is a component that adds no methods and no properties. However, that doesn't mean it
+isn't useful! Tags can be used as names, labels or keywords that group game objects, such as
+enemies, friends, etc.
+
+If you pass a string in the array to `add()`, KAPLAY will automatically
+convert it into an "anonymous component" with the name of the tag.
```
Comment (lajbel) [inline guides/en/1_getting_started/3_components.md]: I would left this as before

File: guides/en/1_getting_started/3_components.md [hunk 5]
```diff
@@ -67,9 +71,7 @@ const bobo = add([
 ]);
 ```
 
-There are many functions that use tags.
-
-For example [`get()`](/doc/ctx/get), to _get_ all game objects with a certain
+There are many functions that look at tags. The simplest one is [`get()`](/doc/ctx/get), which just returns all the game objects with a certain
```
Comment (lajbel) [inline guides/en/1_getting_started/3_components.md]: I prefer as before, simple and short

File: guides/en/1_getting_started/3_components.md [hunk 6]
```diff
@@ -78,10 +80,10 @@ const enemies = get("enemy"); // ghosty and bobo
 
 ## Adding Components and tags
 
-To assign a component or tag to an object, we usually do it on the object's
-creation.
+In most cases components and tags are just added once, when the object is created, by
+putting them in the array passed to `add()`.
```
Comment (lajbel) [inline guides/en/1_getting_started/3_components.md:85]: them in the array passed to `add` => them in `add()`

File: guides/en/1_getting_started/3_components.md [hunk 7]
```diff
@@ -95,11 +97,12 @@ const player = add([
 
 ## Dynamic Components and tags
 
-Usually, you want to assign a component in real time, for example, when the
-player picks up a power-up.
+In some cases (and more commonly with tags than components), you will want to give an existing
+object a new component or tag. For example, when the player picks up a power-up, add a "super" tag,
+and then remove it some time later.
```
Comment (lajbel) [inline guides/en/1_getting_started/3_components.md:103]: As before

File: guides/en/1_getting_started/3_components.md [hunk 8]
```diff
@@ -118,8 +121,11 @@ player.use(sprite("big-butterfly"));
 player.use("big");
 ```
 
-Use is a **use**ful method, but what if you want to remove a component/tag? You
-can use the `GameObjRaw.unuse()` method, passing the component id or tag name.
+*(Aside: the `.use()` method is actually how `add()` creates the object internally. It initializes an empty `GameObj` and then calls `.use()` on all of the components and tags passed in. Don't just take my word for it, [look at the code!](https://github.com/kaplayjs/kaplay/blob/b57385be0f33b89db79ced6a4967d219d240f523/src/game/make.ts#L624-L626))*
```
Comment (lajbel) [inline guides/en/1_getting_started/3_components.md:125]: I like the idea, but looks strange on here, maybe we can create a new component to add these kind of elements? would be cool

Comment (lajbel) [reply guides/en/1_getting_started/3_components.md:125]: So for now remove and save on some gist, will let you know when I add something like that

File: guides/en/1_getting_started/3_components.md [hunk 9]
```diff
@@ -118,8 +121,11 @@ player.use(sprite("big-butterfly"));
...
+
+
+`use()` is a **use**ful method, but what if you want to remove a component or tag? You
+can use the `GameObjRaw.unuse()` method, passing the component ID or tag name. Do note that you can't pass the component itself to `unuse()`, you have to pass the component *name*!
```
Comment (lajbel) [inline guides/en/1_getting_started/3_components.md:129]: The Do note that you can pass the component itself can be in a extra note, as the previous thing. Will let you know then, for now, remove

File: guides/en/1_getting_started/3_components.md [hunk 10]
```diff
@@ -149,38 +155,43 @@ onKeyDown("left", () => {
...
 ]);
 
-console.log(player.c("pos")); // all related state to pos()
+console.log(player.c("pos")); // all the state exclusive to the pos() component
 ```
 
+Note that if a component stores state in a function closure and doesn't expose it as a property or method,
+you still won't be able to access it this way. ¯\\\_(ツ)\_/¯
+(It's a limitation of Javascript, unfortunately. KAPLAY will never be able to get around this.)
+
```
Comment (lajbel) [inline guides/en/1_getting_started/3_components.md:173]: Technical information, doesn't matter for user, save in a gist may reuse later

File: guides/en/1_getting_started/3_components.md [hunk 11]
```diff
@@ -149,38 +155,43 @@ onKeyDown("left", () => {
...
 
 ```js
 if (player.is("enemy")) {
-    debug.log("It is an enemy!");
+    debug.log("EXTERMINATE! EXTERMINATE!");
 }
 ```
 
-This function also works with components, passing the id (the name) of the
+This function also works with components, passing the ID (the name) of the
```
Comment (lajbel) [inline guides/en/1_getting_started/3_components.md:184]: I prefer it lowercase

**Review by lajbel — CHANGES_REQUESTED** [review]

File: guides/en/1_getting_started/3_components.md (hunk 2 above)
Comment (dragoncoder047) [reply guides/en/1_getting_started/3_components.md:29]: ... the what?

File: guides/en/1_getting_started/3_components.md (hunk 8 above)
Comment (dragoncoder047) [reply guides/en/1_getting_started/3_components.md:125]: Ill just comment it out

**Comment by lajbel** [comment]
Will merge and change by myself some stuff, thank yoU!

**Pull request merged**